tests/unit_tests/gebt_poc/test_solver.cpp

```diff
@@ -390,4 +390,490 @@ TEST(SolverTest, CalculateStaticResidualNonZeroValues) {
     );
 }
 
+TEST(SolverTest, CalculateIterationMatrixComponents) {
```

Comment:
I think we might want some "null" tests like we have for the residual matrix.  This will make refactoring a bit easier.